### PR TITLE
Allow "integer" defaults for double and float

### PR DESCRIPTION
**Pull request by da77a**
See https://issues.apache.org/jira/browse/AVRO-2209

### CODE REVIEW

File: lang/c++/impl/Compiler.cc [hunk 1]
```diff
@@ -229,9 +231,13 @@ static GenericDatum makeGenericDatum(NodePtr n,
         assertType(e, json::etLong);
         return GenericDatum(e.longValue());
     case AVRO_FLOAT:
+        if (dt == json::etLong)
```
Comment (thiru-apache) [inline lang/c++/impl/Compiler.cc]: In order to make it consistent with the rest of C++ code, can you add braces?

     if (dt == json::etLong) {
         return GenericDatum(static_cast<float>(e.longValue()));
     }

File: lang/c++/impl/Compiler.cc [hunk 2]
```diff
@@ -229,9 +231,13 @@ static GenericDatum makeGenericDatum(NodePtr n,
...
+            return GenericDatum(static_cast<float>(e.longValue()));
         assertType(e, json::etDouble);
         return GenericDatum(static_cast<float>(e.doubleValue()));
     case AVRO_DOUBLE:
+        if (dt == json::etLong)
+            return GenericDatum(static_cast<float>(e.longValue()));
```
Comment (thiru-apache) [inline lang/c++/impl/Compiler.cc]: Same as above.

**Comment by da77a** [comment]
Done

**Comment by da77a** [comment]
Confirmed working now (both doubles and floats).

**Comment by kojiromike** [comment]
closes #335 
fixes AVRO-2230

**Pull request merged**